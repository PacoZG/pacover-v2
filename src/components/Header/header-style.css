
.header-style {
    @apply bg-gray-600 bg-opacity-80 dark:bg-gray-900 dark:bg-opacity-80 transform duration-500 rounded-3xl shadow-2xl border border-gray-700;
}

.mobile-nav-link {
    @apply text-gray-800 dark:text-blue-200 text-lg font-medium py-2 px-4 w-full text-center transition duration-300 hover:bg-gray-200 dark:hover:bg-gray-800 rounded-md;
}

.desktop-nav-link {
    @apply text-gray-700 dark:text-blue-300 text-lg font-medium px-4 py-2 transition duration-300 hover:text-blue-700 dark:hover:text-blue-400 hover:underline;
}

.mobile-lang-text-style {
    @apply text-gray-900 dark:text-blue-300 text-xl transition transform duration-500 ease-in-out;
}

.pc-lang-text-style {
    @apply text-gray-900 dark:text-blue-300 text-xl transition transform hover:-translate-y-1 hover:scale-110 hover:text-blue-300 duration-500 ease-in-out;
}

@keyframes slideDown {
    from {
        opacity: 0;
        transform: translateY(-10px);
    }
    to {
        opacity: 1;
        transform: translateY(0);
    }
}

.animation-slide-down {
    animation: slideDown 0.3s ease-out forwards;
}
