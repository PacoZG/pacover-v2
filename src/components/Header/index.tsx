'use client'

import React, { useState } from 'react' // No need for useEffect now
import { LinkedinIcon, WhatsappIcon } from 'react-share'
import { Home } from 'lucide-react'
import { Bars3Icon, XMarkIcon } from '@heroicons/react/16/solid'
import { LanguageModel } from '@/utils/language/LanguageModel'
import Toggle from '@/components/Toggle'
import GitHub from '../icons/Github'
import { useTranslations } from 'next-intl'
import { Link } from '@/i18n/navigation'
import './header-style.css'
import { usePathname } from 'next/navigation'

const Header: React.FC = () => {
  const t = useTranslations('Header')
  const [isMobileMenuOpen, setIsMobileMenuOpen] = useState<boolean>(false)
  const useLanguageModel = LanguageModel()

  const pathname = usePathname()

  const toggleMobileMenu = () => {
    setIsMobileMenuOpen(!isMobileMenuOpen)
  }

  return (
    <div className="fixed z-50 w-screen bg-gray-100 dark:bg-gray-800 bg-opacity-90 dark:bg-opacity-90 backdrop-blur-sm transition duration-500 shadow-md">
      {/* Mobile Header */}
      <div className="md:hidden relative flex flex-col ">
        <div className="flex items-center  justify-between px-4 py-2">
          <Link href="/" className="p-2 text-blue-700 dark:text-blue-400">
            <Home className="h-7 w-7" />
          </Link>

          <div className="flex items-center gap-4">
            <Toggle />

            <button
              className="mobile-lang-text-style"
              onClick={useLanguageModel.handleUsersLanguage}
            >
              {useLanguageModel.usersLanguage === 'en' ? 'ESP' : 'ENG'}
            </button>

            <button
              onClick={() => {
                setIsMobileMenuOpen(!isMobileMenuOpen)
              }}
              className="p-2"
            >
              {isMobileMenuOpen ? (
                <XMarkIcon className="h-7 w-7 text-gray-800 dark:text-blue-200" />
              ) : (
                <Bars3Icon className="h-7 w-7 text-gray-800 dark:text-blue-200" />
              )}
            </button>
          </div>
        </div>

        {isMobileMenuOpen && (
          <>
            {/* The full-screen overlay/background button */}
            <button
              id="langBackground-button"
              onClick={() => {
                setIsMobileMenuOpen(!isMobileMenuOpen)
              }}
              className="fixed inset-0 w-screen h-screen bg-black opacity-50 z-40"
            ></button>

            {/* Mobile Menu Content (conditionally rendered and animated) */}
            <div
              id="mobile-menu-content"
              className="absolute top-14 left-0 w-full bg-gray-100 dark:bg-gray-800 bg-opacity-95 dark:bg-opacity-95 shadow-lg flex flex-col items-center py-4 space-y-3 animation-slide-down z-50"
            >
              <Link
                className={
                  pathname.includes('presentation')
                    ? 'active-mobile-nav-link'
                    : 'mobile-nav-link'
                }
                href="/cv/presentation"
                onClick={toggleMobileMenu}
              >
                {t('presentation')}
              </Link>

              <Link
                className={
                  pathname.includes('skills_strenghts')
                    ? 'active-mobile-nav-link'
                    : 'mobile-nav-link'
                }
                href="/cv/skills_strenghts"
                onClick={toggleMobileMenu}
              >
                {t('skills')}
              </Link>

              <Link
                className={
                  pathname.includes('experience')
                    ? 'active-mobile-nav-link'
                    : 'mobile-nav-link'
                }
                href="/cv/experience"
                onClick={toggleMobileMenu}
              >
                {t('experience')}
              </Link>

              <div className="flex items-center gap-4 pt-3 border-t border-gray-300 dark:border-gray-700 w-3/4 justify-center">
                <a
                  href="https://www.linkedin.com/in/francisco-zavala/"
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  <LinkedinIcon size={31} borderRadius={10} />
                </a>

                <a
                  href="https://github.com/PacoZG"
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  <GitHub className="bg-gray-300 h-8 w-8 rounded-full dark:bg-gray-700" />
                </a>

                <a
                  href="https://wa.me/358449888032?text=I%20would%20love%20to%20contact%20you%20for%20a%20possible%20interview."
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  <WhatsappIcon size={31} borderRadius={10} />
                </a>
              </div>
            </div>
          </>
        )}
      </div>

      {/* Desktop Header */}
      <div className="hidden md:flex items-center justify-between px-8 py-3">
        <div className="flex items-center gap-2">
          <Link href="/" className="p-2 text-blue-700 dark:text-blue-400">
            <Home className="h-8 w-8" />
          </Link>

          <Toggle />
        </div>

        <div className="flex items-center gap-2 lg:gap-12">
          <Link className="desktop-nav-link" href="/cv/presentation">
            {t('presentation')}
          </Link>

          <Link className="desktop-nav-link" href="/cv/skills_strenghts">
            {t('skills')}
          </Link>

          <Link className="desktop-nav-link" href="/cv/experience">
            {t('experience')}
          </Link>
        </div>

        <div className="flex items-center gap-4">
          <a
            href="https://www.linkedin.com/in/francisco-zavala/"
            target="_blank"
            rel="noopener noreferrer"
          >
            <LinkedinIcon size={30} borderRadius={10} />
          </a>

          <a
            href="https://github.com/PacoZG"
            target="_blank"
            rel="noopener noreferrer"
          >
            <GitHub className="bg-gray-300 h-8 w-8 rounded-full" />
          </a>

          <a
            href="https://wa.me/358449888032?text=I%20would%20love%20to%20contact%20you%20for%20a%20possible%20interview."
            target="_blank"
            rel="noopener noreferrer"
          >
            <WhatsappIcon size={30} borderRadius={10} />
          </a>

          <button
            className="pc-lang-text-style"
            onClick={useLanguageModel.handleUsersLanguage}
          >
            {useLanguageModel.usersLanguage === 'en' ? 'ESP' : 'ENG'}
          </button>
        </div>
      </div>
    </div>
  )
}

export default Header
