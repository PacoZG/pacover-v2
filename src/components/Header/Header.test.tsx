/* eslint-disabled */
import React from 'react'
import { render } from '@testing-library/react'
import { describe, test, expect, vi } from 'vitest'
import Header from '@/components/Header/index'

window.scrollTo = vi.fn()

vi.mock('next-intl', () => ({
  useTranslations: () => {
    return key => {
      return key
    }
  },
}))
describe('Header component', () => {
  test('matches the snapshot', () => {
    const { container } = render(<Header />)

    expect(container).toMatchSnapshot()
  })
})
