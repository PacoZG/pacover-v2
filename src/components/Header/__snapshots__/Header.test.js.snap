// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Snapshot of the Header component 1`] = `
<div
  className="fixed z-10 w-screen bg-transparent transition duration-500"
>
  <div
    className="header-style relative md:hidden flex flex-row items-center justify-center mx-3 m-2"
  >
    <div
      className="flex justify-center w-full"
    >
      <div>
        <div
          className="block md:hidden"
        >
          <div
            className="flex justify-end items-center space-x-2"
          >
            <div>
              <input
                className="hidden"
                id="toggle"
                name=""
                type="checkbox"
              />
              <span
                className="relative w-14 h-7 flex items-center shadow-lg bg-gray-300 dark:bg-gray-300 rounded-full p-1 cursor-pointer"
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="absolute left-1 h-6 w-6 transition duration-500 text-gray-800 text-opacity-0 dark:text-opacity-100"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10 2a.75.75 0 01.75.75v1.5a.75.75 0 01-1.5 0v-1.5A.75.75 0 0110 2zM10 15a.75.75 0 01.75.75v1.5a.75.75 0 01-1.5 0v-1.5A.75.75 0 0110 15zM10 7a3 3 0 100 6 3 3 0 000-6zM15.657 5.404a.75.75 0 10-1.06-1.06l-1.061 1.06a.75.75 0 001.06 1.06l1.06-1.06zM6.464 14.596a.75.75 0 10-1.06-1.06l-1.06 1.06a.75.75 0 001.06 1.06l1.06-1.06zM18 10a.75.75 0 01-.75.75h-1.5a.75.75 0 010-1.5h1.5A.75.75 0 0118 10zM5 10a.75.75 0 01-.75.75h-1.5a.75.75 0 010-1.5h1.5A.75.75 0 015 10zM14.596 15.657a.75.75 0 001.06-1.06l-1.06-1.061a.75.75 0 10-1.06 1.06l1.06 1.06zM5.404 6.464a.75.75 0 001.06-1.06l-1.06-1.06a.75.75 0 10-1.061 1.06l1.06 1.06z"
                  />
                </svg>
                <div
                  className="w-6 h-6 bg-blue-800 rounded-full shadow-md transition duration-500"
                />
                <svg
                  aria-hidden="true"
                  className="absolute right-1 h-5 w-5 transition duration-500 text-gray-800 text-opacity-100 dark:text-opacity-0"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clipRule="evenodd"
                    d="M7.455 2.004a.75.75 0 01.26.77 7 7 0 009.958 7.967.75.75 0 011.067.853A8.5 8.5 0 116.647 1.921a.75.75 0 01.808.083z"
                    fillRule="evenodd"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
        <div
          className="hidden md:block"
        >
          <div
            className="flex justify-end items-center space-x-2"
          >
            <svg
              aria-hidden="true"
              className="h-6 w-6 transition duration-500 dark:text-gray-300"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 2a.75.75 0 01.75.75v1.5a.75.75 0 01-1.5 0v-1.5A.75.75 0 0110 2zM10 15a.75.75 0 01.75.75v1.5a.75.75 0 01-1.5 0v-1.5A.75.75 0 0110 15zM10 7a3 3 0 100 6 3 3 0 000-6zM15.657 5.404a.75.75 0 10-1.06-1.06l-1.061 1.06a.75.75 0 001.06 1.06l1.06-1.06zM6.464 14.596a.75.75 0 10-1.06-1.06l-1.06 1.06a.75.75 0 001.06 1.06l1.06-1.06zM18 10a.75.75 0 01-.75.75h-1.5a.75.75 0 010-1.5h1.5A.75.75 0 0118 10zM5 10a.75.75 0 01-.75.75h-1.5a.75.75 0 010-1.5h1.5A.75.75 0 015 10zM14.596 15.657a.75.75 0 001.06-1.06l-1.06-1.061a.75.75 0 10-1.06 1.06l1.06 1.06zM5.404 6.464a.75.75 0 001.06-1.06l-1.06-1.06a.75.75 0 10-1.061 1.06l1.06 1.06z"
              />
            </svg>
            <div>
              <input
                className="hidden"
                id="toggle"
                name=""
                type="checkbox"
              />
              <span
                className="w-9 h-5 flex items-center shadow-lg bg-gray-300 dark:bg-gray-300 rounded-full p-1 cursor-pointer"
                onClick={[Function]}
              >
                <div
                  className="w-4 h-4 bg-blue-800 rounded-full shadow-md transition duration-100"
                />
              </span>
            </div>
            <svg
              aria-hidden="true"
              className="h-5 w-5 transition duration-500 dark:text-gray-300 "
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M7.455 2.004a.75.75 0 01.26.77 7 7 0 009.958 7.967.75.75 0 011.067.853A8.5 8.5 0 116.647 1.921a.75.75 0 01.808.083z"
                fillRule="evenodd"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      className="flex items-center gap-4 p-2"
    >
      <a
        href="https://www.linkedin.com/in/francisco-zavala/"
        target="blank"
      >
        <svg
          height={31}
          viewBox="0 0 64 64"
          width={31}
        >
          <rect
            fill="#007fb1"
            height="64"
            rx={10}
            ry={10}
            style={Object {}}
            width="64"
          />
          <path
            d="M20.4,44h5.4V26.6h-5.4V44z M23.1,18c-1.7,0-3.1,1.4-3.1,3.1c0,1.7,1.4,3.1,3.1,3.1 c1.7,0,3.1-1.4,3.1-3.1C26.2,19.4,24.8,18,23.1,18z M39.5,26.2c-2.6,0-4.4,1.4-5.1,2.8h-0.1v-2.4h-5.2V44h5.4v-8.6 c0-2.3,0.4-4.5,3.2-4.5c2.8,0,2.8,2.6,2.8,4.6V44H46v-9.5C46,29.8,45,26.2,39.5,26.2z"
            fill="white"
          />
        </svg>
      </a>
      <a
        href="https://github.com/PacoZG"
        target="blank"
      >
        <svg
          className="bg-gray-300 h-8 w-8 rounded-full"
        >
          github.svg
        </svg>
      </a>
      <a
        href="https://wa.me/358449888032?text=I%20will%20love%20to%20contact%20you%20for%20a%20possible%20interview."
        target="blank"
      >
        <svg
          height={31}
          viewBox="0 0 64 64"
          width={31}
        >
          <rect
            fill="#25D366"
            height="64"
            rx={10}
            ry={10}
            style={Object {}}
            width="64"
          />
          <path
            d="m42.32286,33.93287c-0.5178,-0.2589 -3.04726,-1.49644 -3.52105,-1.66732c-0.4712,-0.17346 -0.81554,-0.2589 -1.15987,0.2589c-0.34175,0.51004 -1.33075,1.66474 -1.63108,2.00648c-0.30032,0.33658 -0.60064,0.36247 -1.11327,0.12945c-0.5178,-0.2589 -2.17994,-0.80259 -4.14759,-2.56312c-1.53269,-1.37217 -2.56312,-3.05503 -2.86603,-3.57283c-0.30033,-0.5178 -0.03366,-0.80259 0.22524,-1.06149c0.23301,-0.23301 0.5178,-0.59547 0.7767,-0.90616c0.25372,-0.31068 0.33657,-0.5178 0.51262,-0.85437c0.17088,-0.36246 0.08544,-0.64725 -0.04402,-0.90615c-0.12945,-0.2589 -1.15987,-2.79613 -1.58964,-3.80584c-0.41424,-1.00971 -0.84142,-0.88027 -1.15987,-0.88027c-0.29773,-0.02588 -0.64208,-0.02588 -0.98382,-0.02588c-0.34693,0 -0.90616,0.12945 -1.37736,0.62136c-0.4712,0.5178 -1.80194,1.76053 -1.80194,4.27186c0,2.51134 1.84596,4.945 2.10227,5.30747c0.2589,0.33657 3.63497,5.51458 8.80262,7.74113c1.23237,0.5178 2.1903,0.82848 2.94111,1.08738c1.23237,0.38836 2.35599,0.33657 3.24402,0.20712c0.99159,-0.15534 3.04985,-1.24272 3.47963,-2.45956c0.44013,-1.21683 0.44013,-2.22654 0.31068,-2.45955c-0.12945,-0.23301 -0.46601,-0.36247 -0.98382,-0.59548m-9.40068,12.84407l-0.02589,0c-3.05503,0 -6.08417,-0.82849 -8.72495,-2.38189l-0.62136,-0.37023l-6.47252,1.68286l1.73463,-6.29129l-0.41424,-0.64725c-1.70875,-2.71846 -2.6149,-5.85116 -2.6149,-9.07706c0,-9.39809 7.68934,-17.06155 17.15993,-17.06155c4.58253,0 8.88029,1.78642 12.11655,5.02268c3.23625,3.21036 5.02267,7.50812 5.02267,12.06476c-0.0078,9.3981 -7.69712,17.06155 -17.14699,17.06155m14.58906,-31.58846c-3.93529,-3.80584 -9.1133,-5.95471 -14.62789,-5.95471c-11.36055,0 -20.60848,9.2065 -20.61625,20.52564c0,3.61684 0.94757,7.14565 2.75211,10.26282l-2.92557,10.63564l10.93337,-2.85309c3.0136,1.63108 6.4052,2.4958 9.85634,2.49839l0.01037,0c11.36574,0 20.61884,-9.2091 20.62403,-20.53082c0,-5.48093 -2.14111,-10.64081 -6.03239,-14.51915"
            fill="white"
          />
        </svg>
      </a>
    </div>
    <div
      className="flex justify-center w-full"
    >
      <button
        className="mobile-lang-text-style"
        onClick={[Function]}
      >
        ESP
      </button>
    </div>
    <div
      className="header-style link-menu"
    >
      <div
        className="mobile-link-group"
      >
        <a
          className="mobile-link"
          href="/cv/presentation"
          onClick={[Function]}
        >
          Presentation
        </a>
        <a
          className="mobile-link"
          href="/cv/skills_strenghts"
          onClick={[Function]}
        >
          Skills
        </a>
        <a
          className="mobile-link"
          href="/cv/experience"
          onClick={[Function]}
        >
          Experience
        </a>
      </div>
    </div>
  </div>
  <div
    className="hidden md:block"
  >
    <div
      className="header-style flex flex-row items-center justify-evenly p-1 shadow-2xl  mr-14 ml-10 m-2"
    >
      <div>
        <div
          className="block md:hidden"
        >
          <div
            className="flex justify-end items-center space-x-2"
          >
            <div>
              <input
                className="hidden"
                id="toggle"
                name=""
                type="checkbox"
              />
              <span
                className="relative w-14 h-7 flex items-center shadow-lg bg-gray-300 dark:bg-gray-300 rounded-full p-1 cursor-pointer"
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="absolute left-1 h-6 w-6 transition duration-500 text-gray-800 text-opacity-0 dark:text-opacity-100"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10 2a.75.75 0 01.75.75v1.5a.75.75 0 01-1.5 0v-1.5A.75.75 0 0110 2zM10 15a.75.75 0 01.75.75v1.5a.75.75 0 01-1.5 0v-1.5A.75.75 0 0110 15zM10 7a3 3 0 100 6 3 3 0 000-6zM15.657 5.404a.75.75 0 10-1.06-1.06l-1.061 1.06a.75.75 0 001.06 1.06l1.06-1.06zM6.464 14.596a.75.75 0 10-1.06-1.06l-1.06 1.06a.75.75 0 001.06 1.06l1.06-1.06zM18 10a.75.75 0 01-.75.75h-1.5a.75.75 0 010-1.5h1.5A.75.75 0 0118 10zM5 10a.75.75 0 01-.75.75h-1.5a.75.75 0 010-1.5h1.5A.75.75 0 015 10zM14.596 15.657a.75.75 0 001.06-1.06l-1.06-1.061a.75.75 0 10-1.06 1.06l1.06 1.06zM5.404 6.464a.75.75 0 001.06-1.06l-1.06-1.06a.75.75 0 10-1.061 1.06l1.06 1.06z"
                  />
                </svg>
                <div
                  className="w-6 h-6 bg-blue-800 rounded-full shadow-md transition duration-500"
                />
                <svg
                  aria-hidden="true"
                  className="absolute right-1 h-5 w-5 transition duration-500 text-gray-800 text-opacity-100 dark:text-opacity-0"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clipRule="evenodd"
                    d="M7.455 2.004a.75.75 0 01.26.77 7 7 0 009.958 7.967.75.75 0 011.067.853A8.5 8.5 0 116.647 1.921a.75.75 0 01.808.083z"
                    fillRule="evenodd"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
        <div
          className="hidden md:block"
        >
          <div
            className="flex justify-end items-center space-x-2"
          >
            <svg
              aria-hidden="true"
              className="h-6 w-6 transition duration-500 dark:text-gray-300"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 2a.75.75 0 01.75.75v1.5a.75.75 0 01-1.5 0v-1.5A.75.75 0 0110 2zM10 15a.75.75 0 01.75.75v1.5a.75.75 0 01-1.5 0v-1.5A.75.75 0 0110 15zM10 7a3 3 0 100 6 3 3 0 000-6zM15.657 5.404a.75.75 0 10-1.06-1.06l-1.061 1.06a.75.75 0 001.06 1.06l1.06-1.06zM6.464 14.596a.75.75 0 10-1.06-1.06l-1.06 1.06a.75.75 0 001.06 1.06l1.06-1.06zM18 10a.75.75 0 01-.75.75h-1.5a.75.75 0 010-1.5h1.5A.75.75 0 0118 10zM5 10a.75.75 0 01-.75.75h-1.5a.75.75 0 010-1.5h1.5A.75.75 0 015 10zM14.596 15.657a.75.75 0 001.06-1.06l-1.06-1.061a.75.75 0 10-1.06 1.06l1.06 1.06zM5.404 6.464a.75.75 0 001.06-1.06l-1.06-1.06a.75.75 0 10-1.061 1.06l1.06 1.06z"
              />
            </svg>
            <div>
              <input
                className="hidden"
                id="toggle"
                name=""
                type="checkbox"
              />
              <span
                className="w-9 h-5 flex items-center shadow-lg bg-gray-300 dark:bg-gray-300 rounded-full p-1 cursor-pointer"
                onClick={[Function]}
              >
                <div
                  className="w-4 h-4 bg-blue-800 rounded-full shadow-md transition duration-100"
                />
              </span>
            </div>
            <svg
              aria-hidden="true"
              className="h-5 w-5 transition duration-500 dark:text-gray-300 "
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M7.455 2.004a.75.75 0 01.26.77 7 7 0 009.958 7.967.75.75 0 011.067.853A8.5 8.5 0 116.647 1.921a.75.75 0 01.808.083z"
                fillRule="evenodd"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        className="flex items-center gap-8 lg:gap-16 p-2"
      >
        <a
          href="https://www.linkedin.com/in/francisco-zavala/"
          target="blank"
        >
          <svg
            height={46}
            viewBox="0 0 64 64"
            width={46}
          >
            <rect
              fill="#007fb1"
              height="64"
              rx={10}
              ry={10}
              style={Object {}}
              width="64"
            />
            <path
              d="M20.4,44h5.4V26.6h-5.4V44z M23.1,18c-1.7,0-3.1,1.4-3.1,3.1c0,1.7,1.4,3.1,3.1,3.1 c1.7,0,3.1-1.4,3.1-3.1C26.2,19.4,24.8,18,23.1,18z M39.5,26.2c-2.6,0-4.4,1.4-5.1,2.8h-0.1v-2.4h-5.2V44h5.4v-8.6 c0-2.3,0.4-4.5,3.2-4.5c2.8,0,2.8,2.6,2.8,4.6V44H46v-9.5C46,29.8,45,26.2,39.5,26.2z"
              fill="white"
            />
          </svg>
        </a>
        <a
          href="https://github.com/PacoZG"
          target="blank"
        >
          <svg
            className="bg-gray-300  h-11 w-11 rounded-full"
          >
            github.svg
          </svg>
        </a>
        <a
          href="https://wa.me/358449888032?text=I%20will%20love%20to%20contact%20you%20for%20a%20possible%20interview."
          target="blank"
        >
          <svg
            height={46}
            viewBox="0 0 64 64"
            width={46}
          >
            <rect
              fill="#25D366"
              height="64"
              rx={10}
              ry={10}
              style={Object {}}
              width="64"
            />
            <path
              d="m42.32286,33.93287c-0.5178,-0.2589 -3.04726,-1.49644 -3.52105,-1.66732c-0.4712,-0.17346 -0.81554,-0.2589 -1.15987,0.2589c-0.34175,0.51004 -1.33075,1.66474 -1.63108,2.00648c-0.30032,0.33658 -0.60064,0.36247 -1.11327,0.12945c-0.5178,-0.2589 -2.17994,-0.80259 -4.14759,-2.56312c-1.53269,-1.37217 -2.56312,-3.05503 -2.86603,-3.57283c-0.30033,-0.5178 -0.03366,-0.80259 0.22524,-1.06149c0.23301,-0.23301 0.5178,-0.59547 0.7767,-0.90616c0.25372,-0.31068 0.33657,-0.5178 0.51262,-0.85437c0.17088,-0.36246 0.08544,-0.64725 -0.04402,-0.90615c-0.12945,-0.2589 -1.15987,-2.79613 -1.58964,-3.80584c-0.41424,-1.00971 -0.84142,-0.88027 -1.15987,-0.88027c-0.29773,-0.02588 -0.64208,-0.02588 -0.98382,-0.02588c-0.34693,0 -0.90616,0.12945 -1.37736,0.62136c-0.4712,0.5178 -1.80194,1.76053 -1.80194,4.27186c0,2.51134 1.84596,4.945 2.10227,5.30747c0.2589,0.33657 3.63497,5.51458 8.80262,7.74113c1.23237,0.5178 2.1903,0.82848 2.94111,1.08738c1.23237,0.38836 2.35599,0.33657 3.24402,0.20712c0.99159,-0.15534 3.04985,-1.24272 3.47963,-2.45956c0.44013,-1.21683 0.44013,-2.22654 0.31068,-2.45955c-0.12945,-0.23301 -0.46601,-0.36247 -0.98382,-0.59548m-9.40068,12.84407l-0.02589,0c-3.05503,0 -6.08417,-0.82849 -8.72495,-2.38189l-0.62136,-0.37023l-6.47252,1.68286l1.73463,-6.29129l-0.41424,-0.64725c-1.70875,-2.71846 -2.6149,-5.85116 -2.6149,-9.07706c0,-9.39809 7.68934,-17.06155 17.15993,-17.06155c4.58253,0 8.88029,1.78642 12.11655,5.02268c3.23625,3.21036 5.02267,7.50812 5.02267,12.06476c-0.0078,9.3981 -7.69712,17.06155 -17.14699,17.06155m14.58906,-31.58846c-3.93529,-3.80584 -9.1133,-5.95471 -14.62789,-5.95471c-11.36055,0 -20.60848,9.2065 -20.61625,20.52564c0,3.61684 0.94757,7.14565 2.75211,10.26282l-2.92557,10.63564l10.93337,-2.85309c3.0136,1.63108 6.4052,2.4958 9.85634,2.49839l0.01037,0c11.36574,0 20.61884,-9.2091 20.62403,-20.53082c0,-5.48093 -2.14111,-10.64081 -6.03239,-14.51915"
              fill="white"
            />
          </svg>
        </a>
      </div>
      <button
        className="pc-lang-text-style"
        onClick={[Function]}
      >
        ESP
      </button>
      <div
        className="header-style web-link-menu"
      >
        <div
          className="icon-container"
        >
          <svg
            aria-hidden="true"
            className="menu-icon"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M2 3.75A.75.75 0 012.75 3h14.5a.75.75 0 010 1.5H2.75A.75.75 0 012 3.75zm0 4.167a.75.75 0 01.75-.75h14.5a.75.75 0 010 1.5H2.75a.75.75 0 01-.75-.75zm0 4.166a.75.75 0 01.75-.75h14.5a.75.75 0 010 1.5H2.75a.75.75 0 01-.75-.75zm0 4.167a.75.75 0 01.75-.75h14.5a.75.75 0 010 1.5H2.75a.75.75 0 01-.75-.75z"
              fillRule="evenodd"
            />
          </svg>
          <svg
            aria-hidden="true"
            className="x-icon"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.28 5.22a.75.75 0 00-1.06 1.06L8.94 10l-3.72 3.72a.75.75 0 101.06 1.06L10 11.06l3.72 3.72a.75.75 0 101.06-1.06L11.06 10l3.72-3.72a.75.75 0 00-1.06-1.06L10 8.94 6.28 5.22z"
            />
          </svg>
        </div>
        <div
          className="pc-link-group"
        >
          <a
            className="link"
            href="/cv/presentation"
            onClick={[Function]}
          >
            Presentation
          </a>
          <a
            className="link"
            href="/cv/skills_strenghts"
            onClick={[Function]}
          >
            Skills
          </a>
          <a
            className="link"
            href="/cv/experience"
            onClick={[Function]}
          >
            Experience
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
